### PR TITLE
add AuthenticationService, JwtAuthenticationFilter, JwtUtil, UserLogi…

### PR DESCRIPTION
**Pull request by Alexander1iubchenko**
…nRequestDto, UserLoginResponseDto, login endpoint to AuthController, used new filter in the SecurityFilterChain bean

### CODE REVIEW

File: src/main/java/controller/AuthController.java [hunk 1]
```diff
@@ -17,10 +20,16 @@
...
+    private final AuthenticationService authenticationService;
 
     @PostMapping("/registration")
     public UserResponseDto register(@RequestBody UserRegistrationRequestDto requestDto)
             throws RegistrationException {
         return userService.register(requestDto);
     }
+
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto request) {
```
Comment (andrii-hoienko) [inline src/main/java/controller/AuthController.java]: Please add @Valid annotation to apply validation rules

File: src/main/java/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (andrii-hoienko) [inline src/main/java/security/JwtAuthenticationFilter.java]: Please don`t forget about constants

File: src/main/java/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (andrii-hoienko) [inline src/main/java/security/JwtAuthenticationFilter.java]: Please make "Bearer " constant

File: src/main/java/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            return bearerToken.substring(7);
```
Comment (andrii-hoienko) [inline src/main/java/security/JwtAuthenticationFilter.java]: magic number

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]

**Review by d1sam — APPROVED** [review]
Well done!

**Pull request merged**